fileserver/routes.py

```diff
@@ -42,6 +56,101 @@ def generate_file_id(data):
         blake2b(data, digest_size=33, salt=b"SessionFileSvr\0\0").digest()
     ).decode()
 
+def abort_with_reason(code, msg, warn=True):
+    if warn:
+        app.logger.warning(msg)
+    else:
+        app.logger.debug(msg)
+    abort(Response(msg, status=code, mimetype='text/plain'))
+
+def valid_blinded_version_id_for_auth(request, required):
+    """
+    Check if a request is correctly authenticated, if the auth headers are missing and auth isn't
+    required then just return 'None'.
+    """
+    pk, ts_str, sig_in = (
+        request.headers.get(f"X-FS-{h}") for h in ('Pubkey', 'Timestamp', 'Signature')
+    )
+    missing = sum(x is None or x == '' for x in (pk, ts_str, sig_in))
+
+    # If we were missing one of the auth headers and don't require auth then just return a 'None'
+    if missing > 0 and not required:
+        return None
+
+    # Parameter input validation
+
+    try:
+        blinded_version_id = pk
+        pk = utils.decode_hex_or_b64(pk, 33)
+    except Exception:
+        abort_with_reason(
+            http.BAD_REQUEST, "Invalid authentication: X-FS-Pubkey is not a valid 66-hex digit id"
+        )
+
+    if pk[0] not in (0x07,):
+        abort_with_reason(
+            http.BAD_REQUEST, "Invalid authentication: X-FS-Pubkey must be 07- prefixed"
+        )
+    pk = pk[1:]
+
+    if not sodium.crypto_core_ed25519_is_valid_point(pk):
+        abort_with_reason(
+            http.BAD_REQUEST,
+            "Invalid authentication: given X-FS-Pubkey is not a valid Ed25519 pubkey",
+        )
+
+    try:
+        sig_in = utils.decode_hex_or_b64(sig_in, 64)
+    except Exception:
+        abort_with_reason(
+            http.BAD_REQUEST, "Invalid authentication: X-FS-Signature is not base64[88]"
```

Comment:
```suggestion
            http.BAD_REQUEST, "Invalid authentication: X-FS-Signature is not base64[86]"
```
(Only 86 b64 chars in a 64-byte value; there *can* be two extra padding bytes to pad it out to 88, but they aren't required).

Reply:
Good catch - updated